src/storage.rs

```diff
@@ -0,0 +1,66 @@
+use alloc::collections::BTreeSet;
+use alloy_primitives::{map::HashMap, Address, B256};
+use revm::{
+    bytecode::opcode,
+    context::ContextTr,
+    inspector::JournalExt,
+    interpreter::{
+        interpreter_types::{InputsTr, Jumps},
+        Interpreter,
+    },
+    Inspector,
+};
+
+/// An Inspector that tracks warm and cold storage slot accesses.
+#[derive(Debug, Default)]
+pub struct StorageInspector {
+    /// Tracks which storage slots have been accessed
+    accessed_slots: HashMap<Address, BTreeSet<B256>>,
+    /// Counter for cold SLOADs
+    cold_loads: u64,
+    /// Counter for warm SLOADs  
+    warm_loads: u64,
```

Comment:
cool, we can track this as part of the address, we can simply convert these two fields into a new struct and convert BTreeSet<B256> into a map